image-annotator/src/redux/reducers/__tests__/faces.test.js

```diff
@@ -0,0 +1,235 @@
+import facesReducer from '../faces';
+
+// Import action labels & action creators for use in specs
+import * as actions from '../../actions';
+
+describe('faces reducer', () => {
+
+  it('is a function', () => {
+    expect(facesReducer).toBeDefined();
+    expect(facesReducer).toBeInstanceOf(Function);
+  });
+
+  it('initializes a default state object', () => {
+    const initialState = facesReducer(undefined, {});
+    expect(initialState).toEqual({
+      list: [],
+      selected: null,
+    });
+  });
+
+  it('does not mutate state when receiving an irrelevant action', () => {
+    const initialState = facesReducer(undefined, {});
+    const nextState = facesReducer(initialState, {});
+    expect(nextState).toBe(initialState);
+  });
+
+  const { RECEIVE_FACES } = actions;
+  describe(`on ${RECEIVE_FACES}`, () => {
+
+    it('sets the action\'s array payload as the faces property', () => {
+      const initialState = facesReducer(undefined, {});
+      const nextState = facesReducer(initialState, {
+        type: RECEIVE_FACES,
+        payload: [{ id: 'a' }, { id: 'b' }, { id: 'c' }],
+      });
+      expect(nextState).not.toBe(initialState);
+      expect(nextState.list).toBeDefined();
+      expect(nextState.list).not.toBe(initialState.list);
+      expect(nextState.list).toEqual([{ id: 'a' }, { id: 'b' }, { id: 'c' }]);
+    });
+
+    it('auto-selects the first face in the set', () => {
+      const initialState = facesReducer(undefined, {});
+      expect(initialState.selected).toBeNull();
+      const nextState = facesReducer(initialState, {
+        type: RECEIVE_FACES,
+        payload: [{ id: 'a' }, { id: 'b' }, { id: 'c' }],
+      });
+      expect(nextState.selected).toBe(0);
+    });
+
+    it('replaces the faces list on subsequent calls', () => {
+      const initialState = facesReducer({
+        selected: 0,
+        list: [{ id: 'a' }, { id: 'b' }, { id: 'c' }],
+      }, {});
+      const nextState = facesReducer(initialState, {
+        type: RECEIVE_FACES,
+        payload: [{ id: 'd' }, { id: 'e' }, { id: 'f' }],
+      });
+      expect(nextState).not.toBe(initialState);
+      expect(nextState.list).toBeDefined();
+      expect(nextState.list).not.toBe(initialState.list);
+      expect(nextState.list).toEqual([{ id: 'd' }, { id: 'e' }, { id: 'f' }]);
+    });
+
+    it('replaces the faces list on subsequent calls even if payload is the same', () => {
```

Comment:
Is this something you really want to assert?  I could see a future optimization undoing this change in order to prevent re-rendering data that didn't change.

Basically, I'd argue that this behavior (though currently what the reducer does) to be "undefined" as of yet.